Source/Core/Core/Core.cpp

```diff
@@ -146,7 +163,16 @@ bool IsRunningInCurrentThread()
 
 bool IsCPUThread()
 {
-	return (s_cpu_thread.joinable() ? (s_cpu_thread.get_id() == std::this_thread::get_id()) : !s_is_started);
+	#ifdef ThreadLocalStorage
+		return tls_is_cpu_thread;
+	#else
+		if(!s_has_init_cpu_tls)
```

Comment:
@magumagu I think I understand. So the init of s_tls_is_cpu_key needs to be put into a void method, and then I need to call that init method from a call to `pthread_once(pthread_once_t, init_for_s_tls_is_cpu_key)`. And that pthread_once() should be called instead of having the s_tls_init_cpu_tls variable.

Right?